run_performance_tests.m

```diff
@@ -0,0 +1,36 @@
+function run_performance_tests(commit)
+    disp(version);
+    if ~exist('spinw', 'class')
+        if exist('swfiles', 'dir') && exist('external', 'dir') && exist('dat_files', 'dir')
+            addpath(genpath('swfiles'));
+            addpath(genpath('external'));
+            addpath(genpath('dat_files'));
+        else
+            error(['SpinW is not installed and the swfiles, external and/or ', ...
+                   'dat_files drectories couldn''t be found on the current ', ...
+                   'path, so the tests cannot be run.'])
+        end
+    end
+    
+    % checkout a hard-coded commit sha for last updated benchmarking result
+    % unless a specific commit (or branch name) is provided.
+    if nargin == 0
+        commit = "b0e60e32aa3790416b65386a721682e5361c1fb3";
+    end
```

Comment:
I think the default should be 'whatever is already checked out' (so don't do any git stuff), rather than a specific hardcoded commit.

We may want to check against historical commits sometimes but the most common use case will be we're on a branch already and want to check how it has affected the performance

Reply:
If we're happy to rely on the user knowing a specific commit sha, then my inclination would be to remove the git stuff entirely from this script - let the user checkout the correct commit in git external to MATLAB. The advantage of keeping the commit sha in the code here is that it documents the last commit benchmarking was done, but this information could also live in the message of commits made to the performance repo (at the minute I've done both, but it would be easy to forget to do the latter and for them to get out of sync).
If we do keep the git stuff in, I should probably run the scripts within a `try/catch` so I can return the user to the commit/branch they started on.

Reply:
Yeah, I agree a record of the spinw commit should be kept in the performance repo (through both a commit message and the naming of the profile results in future).

I'd be happy to remove the git stuff for now, it seems like it could cause some headaches!